torch_xla/csrc/tensor_methods.cpp

```diff
@@ -328,6 +329,14 @@ std::pair<XLATensor, ir::Value> XLATensor::all_to_all(
   return {input.CreateFrom(ir::Value(node, 0)), ir::Value(node, 1)};
 }
 
+std::pair<XLATensor, ir::Value> XLATensor::collective_permute(
+    const XLATensor& input, const ir::Value& token,
+    const std::vector<std::pair<xla::int64, xla::int64>>& source_target_pairs) {
```

Comment:
Still pass by ref.

Reply:
Passing by ref should be fine here right? As long as in `ops/collectivePermute.h/cpp` takes a pass by value vector we won't have the `move` issue. We could also pass by value in `tensor.h` but that will just make another copy of the vector.

`AllToAll` in tensor.h also takes a const ref of vector
```
  static std::pair<XLATensor, ir::Value> all_to_all(                                                                                                                                                                                               
      const XLATensor& input, const ir::Value& token,                                                                                                                                                                                              
      xla::int64 split_dimension, xla::int64 concat_dimension,                                                                                                                                                                                     
      xla::int64 split_count,                                                                                                                                                                                                                      
      const std::vector<std::vector<xla::int64>>& groups); 
```

Reply:
That is wrong as well 😄  Please change that as well.

Reply:
done.. do you mind explaining why is this needed? The vector is created by lambda function in `init_python_bindings.cpp` That address should be valid during the life time of `torch_xla:collective_permute`. When `torch_xla::ir::ops::CollectivePermute` is called, it will make a copy of the vector.

Reply:
It is not **necessary** 😉 
But when you call a constructor which makes a copy of a non trivial data structure, always pass by value.
If the call site cannot relinquish ownership, one copy will be make (as if the ctor would accept by-ref), but if the call site can relinquish, no extra copies will be made.